### PR TITLE
Microapp 12758 icon urls

### PR DESCRIPTION
**Pull request by martin-prilucik**
iconType was missing for some service configuration which could potentially break the logic introduced in MAS when migrating icon -  only relative URL should be stored and exported for LIBRARY icon type
- sapp and metadata files updated for present bundles
- validation implemented

### CODE REVIEW

**Comment by mavocitrix** [comment]
- [ ] Wait for https://issues.citrite.net/browse/MICROAPP-12758 to be deployed to prod before merging this one.

**Comment by martin-prilucik** [comment]
@mavocitrix @krisicaj 
https://issues.citrite.net/browse/MICROAPP-12758 was released, please review this one

File: bundlegen/src/main/java/com/citrix/microapps/bundlegen/bundles/BundlesLoader.java [hunk 1]
```diff
@@ -369,6 +384,30 @@ private static boolean isTranslationValid(String translationChecksum, ModelTrans
...
+        String iconUrl = serviceConfiguration.map(ServiceConfiguration::getIconUrl).orElse(null);
+        String iconType = serviceConfiguration.map(ServiceConfiguration::getIconType).orElse(null);
+        String metadataIconUrl = metadata.map(Metadata::getIconUrl).map(URI::toString).orElse(null);
+        if (isNull(iconUrl) || isNull(iconType)) {
+            issues.add(new ValidationException("Both iconUrl and iconType have to be specified"));
+        }
+        if (!("LIBRARY".equals(iconType)  || "CUSTOM_IMAGE".equals(iconType))) {
+            issues.add(new ValidationException("Unsupported iconType: " + iconType) );
+        }
+        if ("LIBRARY".equals(iconType)  && !isNull(iconUrl)
```
Comment (krisicaj) [inline bundlegen/src/main/java/com/citrix/microapps/bundlegen/bundles/BundlesLoader.java:400]: Maybe define enum for _iconType_.

File: bundlegen/src/main/java/com/citrix/microapps/bundlegen/bundles/BundlesLoader.java [hunk 2]
```diff
@@ -369,6 +384,30 @@ private static boolean isTranslationValid(String translationChecksum, ModelTrans
...
+                && (URI.create(iconUrl).isAbsolute() || !iconUrl.startsWith("exported"))) {
+            issues.add(new ValidationException("LIBRARY type iconUrl must be relative and start with `exported/`"));
```
Comment (krisicaj) [inline bundlegen/src/main/java/com/citrix/microapps/bundlegen/bundles/BundlesLoader.java:402]: There could be used a constant for `exported` prefix.

**Review by krisicaj — APPROVED** [review]

**Pull request merged**